app/src/main/java/com/eis/audioUtility/AudioUtilityManager.java

```diff
@@ -0,0 +1,199 @@
+     * @return The maximum stream volume (real value).
+     */
+    private static int getMaxVolume(Context context, AUMStream stream) {
+        return getAudioManager(context).getStreamMaxVolume(getStream(stream));
+    }
+
+    /**
+     * Sets up the stream Volume, given a certain percentage.
+     *
+     * @param context    The current Context.
+     * @param stream     The chosen stream (should be ALARM, RING or MUSIC).
+     * @param percentage Target volume (expressed in %).
+     * @throws IllegalArgumentException if percentage is not between 0 and 100.
+     */
+    public static void setVolume(Context context, AUMStream stream, int percentage) throws IllegalArgumentException {
+        if (percentage < MIN_PERCENTAGE)
+            throw new IllegalArgumentException(" Your value is too low. Please insert a value between 0 and 100.");
+        if (percentage > MAX_PERCENTAGE)
+            throw new IllegalArgumentException(" Your value is too high. Please insert a value between 0 and 100.");
+        int maxVolume = getMaxVolume(context, stream);
+
+        // Calculate the real value of the new volume
+        int newVolume = maxVolume * percentage;
+        newVolume = Math.round(newVolume / MAX_PERCENTAGE);
+
+        // Sets the Volume.
+        getAudioManager(context).setStreamVolume(
+                getStream(stream),
+                newVolume,
+                AudioManager.FLAG_SHOW_UI
+        );
+
+
+    }
+
+    /**
+     * Sets up the stream Volume to its maximum value (in percentage).
+     *
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     */
+    public static void setVolumeToMax(Context context, AUMStream stream) {
+        setVolume(context, stream, MAX_PERCENTAGE);
+    }
+
+    /**
+     * Sets up the Stream Volume to its minimum value (in percentage).
+     *
+     * @param context The current Context.
+     * @param stream  The chosen stream (should be ALARM, RING or MUSIC).
+     */
+    public static void setVolumeToMin(Context context, AUMStream stream) {
+        setVolume(context, stream, MIN_PERCENTAGE);
+    }
+
+    /**
+     * @param stream The chosen Stream (should be ALARM, RING or MUSIC, otherwise an exception is thrown).
+     * @return The constant representing the Stream.
+     * @throws IllegalArgumentException if stream is not valid (It shouldn't happen, thanks to the enum type).
+     */
+    private static int getStream(AUMStream stream) throws IllegalArgumentException {
+
+        switch (stream) {
+            case ALARM:
+                return AudioManager.STREAM_ALARM;
+            case RING:
+                return AudioManager.STREAM_RING;
+            case MUSIC:
+                return AudioManager.STREAM_MUSIC;
+            default:
+                throw new IllegalArgumentException("\nIllegal Stream. It should be ALARM, RING or MUSIC .");
```

Comment:
Also, is there a reason why this exception starts with a new line command?

Reply:
See the comment above.
I guess I'm gonna remove it.